actions/restore-workspace

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+
+# This action expands a resource to the jenkins jobs directory.
+
+import action_utils
+import grp
+import os
+import pwd
+
+from charmhelpers.core import hookenv, host
+from charmhelpers.payload import archive
+from charms.reactive import is_state
+from shutil import rmtree
+
+# Fail if jenkins is not ready
+if not is_state('jenkins.configured.admin'):
+    error = 'Jenkins is not ready to perform this action.'
+    detail = 'The jenkins.configured.admin state is not set.'
+    action_utils.fail(error, detail)
+
+# Process action params and set vars for later user
+file_path = hookenv.resource_get("workspace")
+wipe = hookenv.action_get("wipe")
+workspace_dir = "/var/lib/jenkins/jobs"
+
+# Stop jenkins
+host.service_stop('jenkins')
+
+# Extract the archive, wiping/re-creating the current workspace dir if needed
+if wipe is "True":
+    rmtree(workspace_dir)
+    os.mkdir(workspace_dir)
+try:
+    archive.extract(file_path, workspace_dir)
+except Exception as e:
+    error = 'Error extracting {}'.format(file_path)
+    action_utils.fail(error, e)
+
+# Make sure our workspace is owned by jenkins
+os.chown(workspace_dir,
+         pwd.getpwnam('jenkins').pw_uid,
+         grp.getgrnam('jenkins').gr_gid)
+
+# Start jenkins
+host.service_start('jenkins')
```

Comment:
The files are being copied with no problems, but the jobs do not register automatically. One way to register new jobs is to send the config.xml through the REST API